skimage/segmentation/_join.py

```diff
@@ -333,8 +332,24 @@ def __getitem__(self, index):
         return out
 
     def __setitem__(self, indices, values):
-        if self._array is None:
-            self._array = self.__array__()
-        self._array[indices] = values
-        self.in_values = np.flatnonzero(self._array)
-        self.out_values = self._array[self.in_values]
+        if np.isscalar(indices):
+            indices = np.atleast_1d(indices)
+        elif isinstance(indices, slice):
+            start = indices.start or 0  # treat None or 0 the same way
+            stop = (indices.stop
+                    if indices.stop is not None
+                    else len(self))
+            step = indices.step
+            indices = np.arange(start, stop, step)
+        if indices.dtype == bool:
+            indices = np.flatnonzero(indices)
```

Comment:
I guess this whole section should be refactored as `indices = _normalize_index(indices)`, since it is repeated exactly in `__getitem__`

Reply:
I hesitated, but I will ;)